pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go

```diff
@@ -307,9 +305,17 @@ func aggregatePodStatus(object *unstructured.Unstructured, aggregatedStatusItems
 			pod.Name, item.ClusterName, temp.Phase)
 	}
 
-	if runningFlag {
+	// More details please refer to: https://github.com/karmada-io/karmada/issues/2137
+	switch {
+	case statusSet.Has(string(corev1.PodFailed)):
+		newStatus.Phase = corev1.PodFailed
+	case statusSet.Has(string(corev1.PodPending)):
+		newStatus.Phase = corev1.PodPending
+	case statusSet.Has(string(corev1.PodRunning)):
 		newStatus.Phase = corev1.PodRunning
-	} else {
+	case statusSet.Has(string(corev1.PodSucceeded)):
+		newStatus.Phase = corev1.PodSucceeded
+	default:
```

Comment:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L2636-L2639

`Unknown `, Deprecated it isn't being set since 2015.

Reply:
ok
/lgtm